src/components/NewsLetter.astro

```diff
@@ -1,19 +1,26 @@
+---
+import dotenv from "dotenv"
+dotenv.config()
+---
+
 <div class="container">
   <div class="newsletter-signup flex flex-col items-center gap-12 rounded-xl md:p-24">
     <h2 class="text-center text-3xl md:text-5xl">JOIN OUR NEWSLETTER</h2>
     <p class="text-xl">Subscribe to our newsletter to receive the latest posts. No spam.</p>
     
-    <form class="grid grid-cols-2 gap-4 items-center" action="POST">
+    <form class="grid grid-cols-2 gap-4 items-center" action={process.env.FORM_SECRET} method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="validate" target="_blank" novalidate>
```

Comment:
The action attribute value will need to be added as an environment variable through the Netlify dashboard. The string can be found in Mailchimp's embedded form builder.